keps/sig-storage/661-statefulset-volume-resize/README.md

```diff
@@ -0,0 +1,1085 @@
+* We'd have a plugin in the [driver's deployment](https://github.com/kubernetes-csi/csi-driver-host-path/blob/master/deploy/kubernetes-1.24/hostpath/csi-hostpath-plugin.yaml) that would automatically populate the `CSIDriver` object with the capabilities of the driver by calling `GetPluginCapabilities`.
+  * We could also modify one of the existing plugins (eg `csi-resizer`, `csi-provisioner`) - but none of them looks like a good candidate for this purpose.
+
+The two options have the same drawbacks
+
+* It will require modifying existing CSI drivers to support this feature.
+* I noticed that some drivers don't explicitly specify support for online expansion when calling `GetPluginCapabilities`. Here's an example from the [EBS CSI Driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/7db7d1a745abd1473eb86e63383c119209dcfc80/pkg/driver/identity.go#L36)
+
+Given the above, this kind of validation won't be handled as part of this KEP.
+
+#### Validating That The Volume Size Doesn't Decrease
+
+We don't want to prevent users from decreasing the size of the volume in the StatefulSet PVC templates. Because, we want to support recovery from expansion failures as discussed in [KEP 1790](../1790-recover-resize-failure/README.md). 
+
+However, we'd ideally want to prevent users from decreasing the size of the StatefulSet PVC templates below (or equal) to the current capacity of the volume (`.status.capacity.storage`). 
+This is similar to a validation that performed by the [PVC API server validation code](https://github.com/kubernetes/kubernetes/blob/d89d5ab2680bc74fe4487ad71e514f4e0812d9ce/pkg/apis/core/validation/validation.go#L2256) today. 
+
+There are two problems with performing this kind of validation during StatefulSet admission though.
+
+* The validation logic will have to be informed of and access all of the different PVC objects that exist for the StatefulSet. This might involve a lot of logic that is owned by the StatefulSet controller today (eg PVC naming convensions, ordinals, etc)
+* There could be a race condition where at the time of admission, we see the requested size as being less than the capacity of a PVC - but by the time that the StatefulSet controller tries to reconcole the PVC's request size to that of the template, the capabity of the PVC has already increased.
+
+Given the above, this kind of validation won't be handled as part of this KEP.
+
+* That being said, At the time of writing this document, I believe that [KEP 1790](../1790-recover-resize-failure/README.md) is still disabled behind a feature flag. In that case, one validation that we can do (and will do if it's still the case during implementation), is validate that as long as the feature flag for [KEP 1790](../1790-recover-resize-failure/README.md) is disabled, the request size of the modified PVC template cannot be less than the request size of the old PVC template.
+
```

Comment:
Will this be a validation or done via admission controller? We have to be pretty clear about where we intend to validate this stuff.

Reply:
this can be done in validation since it only needs to look at the statefulset object. will clarify